From 92f27d80d517b84e9db18c2c03b67179050b71b7 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Fri, 8 Jun 2018 13:41:16 +0200
Subject: [PATCH 5/5] TelephonyComponentFactory: Fix invalid phone creation
 another way

* It turns out that the qti-telephony-common.jar needs CDMA_LTE
  for certain devices, so shuffle the hack used in 0497c6b35b0fcb2f008e0c4e7882aa1f9d06079f
  to TelephonyComponentFactory instead of PhoneFactory

Change-Id: Ie9019ed414bdc8d2f5a38a099a2d270de831626d
---
 .../internal/telephony/TelephonyComponentFactory.java  | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/src/java/com/android/internal/telephony/TelephonyComponentFactory.java b/src/java/com/android/internal/telephony/TelephonyComponentFactory.java
index c8f78b91f..742daf536 100644
--- a/src/java/com/android/internal/telephony/TelephonyComponentFactory.java
+++ b/src/java/com/android/internal/telephony/TelephonyComponentFactory.java
@@ -200,15 +200,11 @@ public class TelephonyComponentFactory {
             TelephonyComponentFactory telephonyComponentFactory) {
         Rlog.d(LOG_TAG, "makePhone");
         Phone phone = null;
-        if (precisePhoneType == PhoneConstants.PHONE_TYPE_GSM) {
+        if (precisePhoneType == PhoneConstants.PHONE_TYPE_GSM ||
+                    precisePhoneType == PhoneConstants.PHONE_TYPE_CDMA_LTE) {
             phone = new GsmCdmaPhone(context,
                 ci, notifier, phoneId,
-                PhoneConstants.PHONE_TYPE_GSM,
-                telephonyComponentFactory);
-        } else if (precisePhoneType == PhoneConstants.PHONE_TYPE_CDMA) {
-                phone = new GsmCdmaPhone(context,
-                ci, notifier, phoneId,
-                PhoneConstants.PHONE_TYPE_CDMA_LTE,
+                precisePhoneType,
                 telephonyComponentFactory);
         }
         return phone;
-- 
2.17.1

